src/jit/utils.h

```diff
@@ -42,6 +42,22 @@ inline bool isPow2(T i)
     return (i > 0 && ((i - 1) & i) == 0);
 }
 
+inline bool isPow2f(float v)
+{
+    uint32_t i = *(uint32_t*)&v;
+    uint8_t exponent = (uint8_t)(i >> 23);
+    uint32_t mantissa =  i & 0x7fffff;
+    return mantissa == 0 && exponent != 0 && exponent != 127;
+}
+
+inline bool isPow2d(double v)
+{
+    uint64_t i = *(uint64_t*)&v;
+    uint32_t exponent = (uint32_t)(i >> 52);
+    uint64_t mantissa =  i & 0x8ffffffffffff;
+    return mantissa == 0 && exponent != 0 && exponent != 1023;
```

Comment:
I think this should just return `true` if it is a power of two, even for denormals. The actual optimization code should then filter out denormal inputs and denormal reciprocals separately.

Reply:
Actually this is correct as it is, it's only the function name that is kind of unfortunate. It's not that useful to know that a FP value is a power of 2, it's not like we're going to use shifts, bittests and similar stuff to manipulate FP values. What we're after is basically a precise reciprocal so a `hasPreciseReciprocal` or something like that would be preferable.

Also, this should be moved to `FloatingPointUtils`.